src/packagedcode/dockerfileLABEL.py

```diff
@@ -0,0 +1,72 @@
+#
+# Copyright (c) nexB Inc. and others. All rights reserved.
+# ScanCode is a trademark of nexB Inc.
+# SPDX-License-Identifier: Apache-2.0
+# See http://www.apache.org/licenses/LICENSE-2.0 for the license text.
+# See https://github.com/nexB/scancode-toolkit for support or download.
+# See https://aboutcode.org for more information about nexB OSS projects.
+#
+
+
+
+import io
+from pathlib import Path
+from dockerfile_parse import DockerfileParser  
+from packagedcode import models
+from packagedcode import utils
+
+
+class DockerfileHandler(models.DatafileHandler):
+    datasource_id = 'dockerfile'
+    default_package_type = 'docker-image'
+    path_patterns = ('Dockerfile', 'containerfile', '*.dockerfile')  
+    description = 'Dockerfile (OCI) metadata handler'
+    documentation_url = 'https://docs.docker.com/engine/reference/builder/'
+
+    @classmethod
+    def parse(cls, location, package_only=False):
+        """
+        Parse a Dockerfile and yield one or more PackageData objects with OCI labels and metadata.
+        """
+        labels = cls.extract_oci_labels_from_dockerfile(location)
+
+       
+        package_data = {
+            'datasource_id': cls.datasource_id,
+            'type': cls.default_package_type,
+            'name': labels.get('name', 'unknown'),  

```

Comment:
There is no point having the defaults as `unknown`, it's better to have these as `None` which is what all other parsers do too, if there is no data found, that attribute is just kept empty. This is since we use existence of a value in these attributes for other functions.